autogpt_platform/backend/backend/data/block.py

```diff
@@ -166,6 +170,20 @@ def get_required_fields(cls) -> set[str]:
             if field_info.is_required()
         }
 
+    @classmethod
+    def get_provider_from_credentials_field(cls, field_name: str) -> str | None:
+        """Extract provider name from a credentials field's schema."""
+        field_info = cls.model_fields.get(field_name)
+        if not field_info:
+            return None
+
+        json_schema_extra = field_info.json_schema_extra
+        if isinstance(json_schema_extra, dict):
+            providers = json_schema_extra.get("credentials_provider", [])
+            if providers and isinstance(providers, list):
+                return str(providers[0])
+        return None
```

Comment:
Could be:
```suggestion
        return cls.get_credentials_fields_info()[field_name].provider
```
although this isn't as forgiving to invalid `field_name` inputs

Reply:
This change completely breaks the system, its needs to be very forgiving so that the default is essentially enabling blocks